reach4help/reach4help
# Refactor DeleteUserData Cloud Function to use Batch Writes and Transactions

## Description
Deleting/updating user data fields should be done in an atomic operation.
Batch writes for deleteUserPrivilegedInfo() and deleteUserTimelines().
Transactions for deletePinUserRequests() and deleteCavUserOffersAndRequests().

Tested and works as expected on my local firebase.
## Motivation
Ensures all fields are in same state when deleting/updating data in case of operation failure.
## Testing Guidelines
Deploy cloud function, navigate to settings page, and delete account. Check logs and the function should return 200 status code.
## Release Checklist


## Security Checklist

This PR creates, modifies, or deletes:




## Additional Notes
